presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeWindows.java

```diff
@@ -209,111 +224,156 @@ public void testIdenticalWindowSpecificationsDefaultFrame()
                 "FROM lineitem WHERE (partkey = 272 OR partkey = 273) AND suppkey > 50 " +
                 "), " +
                 "bar AS ( " +
                 "SELECT " +
                 "suppkey, orderkey, partkey, " +
-                "AVG(quantity) OVER (PARTITION BY orderkey ORDER BY shipdate ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) b " +
+                "AVG(quantity) OVER (PARTITION BY orderkey ORDER BY shipdate, quantity DESC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) b " +
                 "FROM lineitem WHERE (partkey = 272 OR partkey = 273) AND suppkey > 50 " +
                 ")" +
                 "SELECT * FROM foo, bar WHERE foo.a = bar.b";
 
+        ExpectedValueProvider<WindowNode.Specification> leftSpecification = specification(
+                ImmutableList.of(ORDERKEY_ALIAS),
+                ImmutableList.of(SHIPDATE_ALIAS, QUANTITY_ALIAS),
+                ImmutableMap.of(SHIPDATE_ALIAS, SortOrder.ASC_NULLS_LAST, QUANTITY_ALIAS, SortOrder.DESC_NULLS_LAST));
+
+        ExpectedValueProvider<WindowNode.Specification> rightSpecification = specification(
+                ImmutableList.of(rOrderkeyAlias),
+                ImmutableList.of(rShipdateAlias, rQuantityAlias),
+                ImmutableMap.of(rShipdateAlias, SortOrder.ASC_NULLS_LAST, rQuantityAlias, SortOrder.DESC_NULLS_LAST));
+
+        ImmutableMap.Builder<String, String> leftBuilder = ImmutableMap.builder();
+        leftBuilder.put(DISCOUNT_ALIAS, "discount");
+        leftBuilder.put(ORDERKEY_ALIAS, "orderkey");
+        leftBuilder.put("PARTKEY", "partkey");
+        leftBuilder.put(SUPPKEY_ALIAS, "suppkey");
+        leftBuilder.put(QUANTITY_ALIAS, "quantity");
+        leftBuilder.put(SHIPDATE_ALIAS, "shipdate");
+
+        PlanMatchPattern leftTableScan = tableScan("lineitem", leftBuilder.build());
+
+        ImmutableMap.Builder<String, String> rightBuilder = ImmutableMap.builder();
+
+        rightBuilder.put(rOrderkeyAlias, "orderkey");
+        rightBuilder.put("R_PARTKEY", "partkey");
+        rightBuilder.put("R_SUPPKEY", "suppkey");
+        rightBuilder.put(rQuantityAlias, "quantity");
+        rightBuilder.put(rShipdateAlias, "shipdate");
+
+        PlanMatchPattern rightTableScan = tableScan("lineitem", rightBuilder.build());
+
         assertUnitPlan(sql,
                 anyTree(
-                        join(JoinNode.Type.INNER, ImmutableList.of(),
-                                any(
-                                        window(ImmutableList.of(functionCall("sum", "*")),
-                                                anyTree())),
-                                any(
-                                        window(ImmutableList.of(functionCall("avg", "*")),
-                                                anyTree())))));
+                        filter("SUM = AVG",
+                                join(JoinNode.Type.INNER, ImmutableList.of(),
+                                        any(
```

Comment:
Ah, and you wanted to track which WF was on which side of the join/that that was handled properly. right?